Remove Zendesk component from Learner Dashboard (fixes #519)

paulbert opened the pull request:
- Deletes the ZendeskFab component
- Removes the mock for ZendeskFab from App.test.jsx
- Removes `react-zendesk` package

I'm not sure about the "Update Jest snapshot" part but I ran the tests locally and they all passed.

Also there are references to a `ZENDESK_KEY` environment variable that don't seem to be used anywhere. Should I remove that in this PR?

Issue references:
Fixes #519 #496 #225

openedx-webhooks commented:
Thanks for the pull request, @paulbert!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@2U-aperture`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

jsnwesson commented:
Hey @paulbert , thanks for picking this up! I forgot to add the environment variable, but that should also be removed in this work, thanks! 

For the Jest snapshots, I realize now that the only snapshots that existed for this component were in the `Zendesk` directory, so I don't think you need to do anything else for that. 

Once you've got the PR up-to-date with the main branch and the env variable is removed, my team should be able to review this PR next week!

paulbert commented:
@jsnwesson Updated to the main branch last night and removed the env variables. I squashed the commits and rewrote the message since I didn’t follow the commit message guidelines the first time. Let me know if a different commit type than refactor would be more appropriate. Thanks!

deborahgu commented:
@jsnwesson when you do sign off on this one, I'm pretty sure that failing codecov  test is just because the removed code and tests change the code:test ratio in the repository, so the percentage report thinks we have a lower code coverage  ratio; this is just removals and there's no actual missing tests.

jsnwesson commented:
Thanks @paulbert ! I use [OEP-51](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html) as a reference for my commits, and so I think `refactor` is correct, although if you can also add a `!` to the commit (ie. `refactor!`), that would make it clear for anyone in the community who has also been using Zendesk that this will be a breaking change for them. Other than that, this looks good to me and I approve it! 

Once you change the commit message, I'll merge it to main :D

paulbert commented:
@jsnwesson Thanks! Updated the commit message and rebased it to be up to date.
